File: 30-07-2025_Subarrays_with_sum_k.java
public int cntSubarrays(int[] arr, int k) {
    int n = arr.length;
    Map<Integer, Integer> mp = new HashMap<>();
    int sum = 0, count = 0;
    mp.put(0, 1);

    for (int i = 0; i < n; i++) {
        sum += arr[i];

        if (mp.containsKey(sum - k)) {
            count += mp.get(sum - k);
        }

        mp.put(sum, mp.getOrDefault(sum, 0) + 1);
    }

    return count;
}

File: 30-07-2025_Subarrays_with_sum_k.cpp
int cntSubarrays(vector<int> &arr, int k) {
    int n = arr.size();
    map<int, int> mp;
    int sum = 0;
    int count = 0;
    mp[0] = 1; 
    for (int i = 0; i < n; i++) {
        sum+=arr[i];

        if (mp.find(sum - k) != mp.end()) {
            count+=mp[sum - k];
        }

        mp[sum]++;
    }

    return count;
}
